ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoice.java

```diff
@@ -582,6 +578,27 @@ private BlockImportResult importBlockAndState(
     return result;
   }
 
+  // from consensus-specs/fork-choice:
+  // get_current_slot(store) == block.slot and is_before_attesting_interval and is_first_block
+  private boolean shouldApplyProposerBoost(
+      final SignedBeaconBlock block, final StoreTransaction transaction) {
+    if (!spec.getCurrentSlot(transaction).equals(block.getSlot())) {
+      return false;
+    }
+
+    final UInt64 millisPerSlot = spec.getMillisPerSlot(block.getSlot());
+    final UInt64 timeIntoSlotMillis = getMillisIntoSlot(transaction, millisPerSlot);
+    if (!isBeforeAttestingInterval(millisPerSlot, timeIntoSlotMillis)) {
+      return false;
+    }
+
+    if (forkChoiceProposerBoostUniquenessEnabled) {
+      return transaction.getProposerBoostRoot().isEmpty();
+    }
+
+    return true;
+  }
```

Comment:
NIT: I like this idea, but maybe we can have the comment on top of each condition?

e.g.

```
// get_current_slot(store) == block.slot
if (!spec.getCurrentSlot(transaction).equals(block.getSlot())) {

[...]

// is_before_attesting_interval
if (!isBeforeAttestingInterval(millisPerSlot, timeIntoSlotMillis)) {

[...]

// is_first_block
if (forkChoiceProposerBoostUniquenessEnabled) {
  return transaction.getProposerBoostRoot().isEmpty();
}
```

Reply:
I like it. Done.